application/clicommands/ScanCommand.php

```diff
@@ -58,9 +70,44 @@ public function indexAction()
                 $name
             );
 
+            $sinceLastScan = $this->processThresholdToDateTiime($this->params->get('since-last-scan'));
+
+            if ($sinceLastScan !== null) {
+                $this->cleanupLastScanTargets($this->getDb(), $sinceLastScan);
+            }
+
             $verified = CertificateUtils::verifyCertificates($this->getDb());
 
+            $sinceLastSeen = $this->processThresholdToDateTiime($this->params->get('since-last-seen'));
+
+            if ($sinceLastSeen !== null) {
+                $this->cleanupLastSeenTargets($this->getDb(), $sinceLastSeen);
+                CertificateUtils::cleanupCertificates($this->getDb(), $sinceLastSeen);
+            }
+
+            CertificateUtils::cleanupCertificateChains($this->getDb());
             Logger::info("Checked %d certificate chain%s.", $verified, $verified !== 1 ? 's' : '');
         }
     }
+
+    protected function cleanupLastScanTargets(Connection $db, DateTime $sinceLastScan)
+    {
+        $db->delete(
+            'x509_target',
+            ['last_scan < ?' => $sinceLastScan->getTimestamp()]
+        );
+    }
+
+    protected function cleanupLastSeenTargets(Connection $db, DateTime $sinceLastSeen)
+    {
+        $db->delete(
+            'x509_target',
+            [
+                new Expression(sprintf(
+                    "last_seen <= last_scan - (UNIX_TIMESTAMP() - %d)",
```

Comment:
Please use single quotes and I think our `Expression` class does also support sprtfing 😆 by itself.